### PR TITLE
Update typespecs and enforce with and limit in Inspect.Algebra

### PR DESCRIPTION
**Pull request by eksperimental**
Additionally it rephrases docs.
************************

I made a typo in `:infinity` and it was giving a cryptic error not pointing me to what was wrong. ArithmeticError

### CODE REVIEW

File: lib/elixir/lib/inspect/algebra.ex [hunk 1]
```diff
@@ -31,30 +31,32 @@ defmodule Inspect.Opts do
...
-    * `:inspect_fun` (since v1.9.0) - a function to build algebra documents,
-      defaults to `Inspect.inspect/2`.
+    * `:inspect_fun` (since v1.9.0) - a function to build algebra documents.
+      Defaults to `Inspect.inspect/2`.
 
     * `:limit` - limits the number of items that are inspected for tuples,
-      bitstrings, maps, lists and any other collection of items. It does not
-      apply to printable strings nor printable charlists and defaults to 50.
+      bitstrings, maps, lists and any other collection of items. It neither
+      applies to printable strings nor to printable charlists.
```
Comment (josevalim) [inline lib/elixir/lib/inspect/algebra.ex:39]: Is the previous sentence incorrect? I am not a native speaker, so I always struggle with neither. I am wondering if there is a clearer way to write this?

Comment (eksperimental) [reply lib/elixir/lib/inspect/algebra.ex:39]: nor is usually used after neither.

This one reads better IMO.

* `:limit` - limits the number of items that are inspected for tuples, 
bitstrings, maps, lists and any other collection of items, with the exception of 
printable strings and printable charlists.

Comment (josevalim) [reply lib/elixir/lib/inspect/algebra.ex:39]: I love it. :+1: I would even add:

> with the exception of printable strings and printable charlists which use the `:printable_limit` option

WDYT?

Comment (eksperimental) [reply lib/elixir/lib/inspect/algebra.ex]: Yeah.. it looks great. pushed!

File: lib/elixir/lib/inspect/algebra.ex [hunk 2]
```diff
@@ -900,7 +906,12 @@ defmodule Inspect.Algebra do
...
 
-  @spec fits?(width :: integer(), column :: integer(), break? :: boolean(), entries) :: boolean()
+  @spec fits?(
+          width :: non_neg_integer(),
+          column :: non_neg_integer(),
+          break? :: boolean(),
+          entries
+        ) :: boolean()
         when entries:
                maybe_improper_list({integer(), mode(), t()}, {:tail, boolean(), entries} | [])
```
Comment (eksperimental) [inline lib/elixir/lib/inspect/algebra.ex:916]: I  am not sure what this interger() is but you could tell me if is should be non_neg_integer() or not.

**Comment by josevalim** [comment]
Please note it broke the Logger test suite. :)

**Comment by eksperimental** [comment]
> Please note it broke the Logger test suite. :)

Doesn't seem easy to solve after tinkering with it for a while, and I have never got my hands into Logger.
Do you think you could see if you can fix it?

**Comment by josevalim** [comment]
@eksperimental please replace Logger.Utils.inspect_data by this:

```elixir
  defp inspect_data([data | _], opts) do
    width = if opts.width == :none, do: 80, else: opts.width

    data
    |> Inspect.Algebra.to_doc(opts)
    |> Inspect.Algebra.format(width)
  end
```

**Comment by eksperimental** [comment]
done!

**Comment by josevalim** [comment]
:green_heart: :blue_heart: :purple_heart: :yellow_heart: :heart:

**Pull request merged**